src/client.ts

```diff
@@ -60,21 +60,27 @@ class Client {
     // Webhooks page for a configured web-hook.
     const delivery: string = data['x-github-delivery']
     const event: string = data['x-github-event']
-    let action: string
 
-    if (body.action === undefined) {
+    let action: string = ''
+    if (body.action !== undefined) {
+      action = body.action
+    } else if (body.payload !== undefined) {
       let payload: any
       try {
         payload = JSON.parse(body.payload)
+        action = payload.action
       } catch (err) {
-        this.logger.error(ts, 'ERROR: bad payload', err, 'body: ', body)
-        return
+        this.logger.warn(ts, 'ERROR: bad payload', err, 'msg: ', msg)
       }
```

Comment:
Since you don't access `payload` outside the block started at line 69, you can move the declaration inside that block and change it to `const`...or dispense with it altogether.
```typescript
    } else if (body.payload !== undefined) {
      try {
        action = JSON.parse(body.payload).action
      } catch (err) {
        this.logger.warn(ts, 'ERROR: bad payload', err, 'msg: ', msg)
      }
```